File: ViewBase/Source/Converters/IntegerStringConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Public Class IntegerStringConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If IsNumeric(value) Then
            Dim int = CType(value, Integer)
            If parameter IsNot Nothing Then
                Return int.ToString(parameter.ToString)
            End If
            Return int.ToString()
        End If
        Return "0"
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        If IsNumeric(value) Then
            Dim str = CType(value, String)
            Dim integerValue As Integer
            Dim ok = Integer.TryParse(str, NumberStyles.Any, CultureInfo.CurrentCulture, integerValue)
            If ok Then
                Return integerValue
            End If
        End If
        Return 0
    End Function
End Class

File: BaseView/src/Converters/IntegerStringConverter.cs
using Microsoft.VisualBasic;
using System;
using System.Globalization;
using System.Windows.Data;
using Wells.Base;

namespace Wells.BaseView.Converters
{
    public class IntegerStringConverter : IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (Common.IsIntegerNumericType(value.GetType()))
            {
                var integer = Convert.ToInt32(value, culture);
                if (parameter != null) { return integer.ToString(parameter.ToString(), culture); }
                return integer.ToString(culture);
            }
            return "0";
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (Information.IsNumeric(value))
            {
                var str = (string)value;
                bool ok = int.TryParse(str, NumberStyles.Any, culture, out int integer);
                if (ok) { return integer; }
            }
            return 0;
        }
    }
}
